test/functorch/test_aotdispatch.py

```diff
@@ -841,8 +841,10 @@ def f(x):
 
         a = torch.ones(4, requires_grad=True)
         a2 = a.clone().detach().requires_grad_()
+        a3 = a.clone().detach().requires_grad_()
+        a4 = a.clone().detach().requires_grad_()
         aa = TwoTensor(a, a2)
-        aa2 = aa.clone().detach().requires_grad_()
+        aa2 = TwoTensor(a3, a4)
```

Comment:
One difference between calling recursively back to `meta_tensor` and using `callback` is that using `meta_tensor` we now faithfully replicate the requires_grad of original tensor.

With this change, this test fails during fakification when constructing the outer TwoTensor out of the two Tensors because the requires_grad mismatch. 

TwoTensor's requires_grad is determined by the requires_grad of the inner tensor, which for aa2 have actually been detached. 

The reason this succeeds in eager because we manually set aa2's requires_grad to True.

Reply:
tbh I think we should fix this on the TwoTensor side - it's a bit of an anti-pattern that I originally wrote `TwoTensor` to "inherit" its requires_grad from the inner tensor. We should just:

(1) not set requires_grad in anyway within the `TwoTensor()` constructor

(2) add a differentiable constructor for TwoTensor that wraps the `TwoTensor()` constructor into an autograd.Function`. e.g. `TwoTensor.from_tensors(a, b)`

does that sound good to you? (it's also more inline with what we do for float8/dtensor)

Reply:
Yeah, inheriting requires grad from inner is just wrong, don't do that lol

Reply:
> not set requires_grad in anyway within the TwoTensor() constructor

Sounds good! There seem to be some places in the tests to update however, so I'm going to do it in a follow up to keep this PR small.